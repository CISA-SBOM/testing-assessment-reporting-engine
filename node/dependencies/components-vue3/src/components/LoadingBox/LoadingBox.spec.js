import { describe, expect, it } from 'vitest'
import { shallowMount } from '@vue/test-utils'
import Component from "./LoadingBox.vue";


describe("LoadingBox.vue", () => {
  it("is a Vue instance", () => {
    const wrapper = shallowMount(Component);
    expect(wrapper.vm).toBeTruthy();
  });
  it("matches snapshot", () => {
    const wrapper = shallowMount(Component);
    expect(wrapper.element).toMatchSnapshot();
  });
  it("matches snapshot with no props assigned", () => {
    const propsData = {};
    const wrapper = shallowMount(Component, { propsData });
    expect(wrapper.html()).toMatchSnapshot();
  });
  it("matches snapshot with width prop assigned", () => {
    const propsData = {
      width: 'w-1/2',
    };
    const wrapper = shallowMount(Component, { propsData });
    expect(wrapper.html()).toMatchSnapshot();
  });
  it("matches snapshot with width and height prop assigned", () => {
    const propsData = {
      width: 'w-1/2',
      height: 'h-32'
    };
    const wrapper = shallowMount(Component, { propsData });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
