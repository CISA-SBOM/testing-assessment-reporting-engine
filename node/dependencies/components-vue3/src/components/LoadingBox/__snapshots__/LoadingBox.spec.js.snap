// Vitest Snapshot v1

exports[`LoadingBox.vue > matches snapshot 1`] = `
<span
  class="loadingBox rounded flex flex-row relative overflow-hidden bg-gray-200 w-full h-full"
  data-id="sds-loading-box"
  data-v-73fb2d42=""
/>
`;

exports[`LoadingBox.vue > matches snapshot with no props assigned 1`] = `"<span data-id=\\"sds-loading-box\\" class=\\"loadingBox rounded flex flex-row relative overflow-hidden bg-gray-200 w-full h-full\\" data-v-73fb2d42=\\"\\"></span>"`;

exports[`LoadingBox.vue > matches snapshot with width and height prop assigned 1`] = `"<span data-id=\\"sds-loading-box\\" class=\\"loadingBox rounded flex flex-row relative overflow-hidden bg-gray-200 w-1/2 h-32\\" data-v-73fb2d42=\\"\\"></span>"`;

exports[`LoadingBox.vue > matches snapshot with width prop assigned 1`] = `"<span data-id=\\"sds-loading-box\\" class=\\"loadingBox rounded flex flex-row relative overflow-hidden bg-gray-200 w-1/2 h-full\\" data-v-73fb2d42=\\"\\"></span>"`;
